### PR TITLE
Milestone 01: Awesome books: plain JavaScript with objects

### PR DESCRIPTION
**Pull request by mavericks-db**
## Awesome books: plain JavaScript with objects

This branch includes:
- [ ] Implement only a basic UI with plain HTML.
- [ ] Create a collection that keeps a list of books.
- [ ] Create a function to add a new book to the collection, with title and author.
- [ ] Create a function to remove a book from the collection.
- [ ] Display all books saved in the collection in the top part of the page.
- [ ] Make sure that when a user clicks the "Add" button:
- A new book is added to the collection.
- The new book is displayed on the page.
- [ ] Make sure that when a user clicks the "Remove" button:
- The correct book is removed from the collection.
- The correct book disappears from the page.
- [ ] Make sure that data is preserved in the browser's memory by using localStorage.
- [ ] Don't use any library or framework, use only plain JavaScript.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const addBtn = document.querySelector('#addBtn');
+const title = document.querySelector('#title');
+const author = document.querySelector('#author');
+const collection = document.querySelector('.collection');
+
+let books = [];
+
```
Comment (ShadyShawkat) [inline script.js:7]: - [ ] Kindly make sure the books array is stored and managed through your local storage, [as per this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m1_plain_js_objects.md#:~:text=Make%20sure%20that%20data%20is%20preserved%20in%20the%20browser%27s%20memory%20by%20using%20localStorage.).

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 1 above)
Comment (mavericks-db) [reply script.js:7]: Hi @ShadyShawkat we stored the book entries in the localStorage. May I know what do you mean on managed through the localStorage?

Comment (ShadyShawkat) [reply script.js:7]: Hey @mavericks-db
By "managed", I mean that when I add some books and refresh the page, I'll still have those books displayed on the page and I can add or remove them. You can use local storage to achieve this behavior.

Comment (mavericks-db) [reply script.js:7]: Hi @ShadyShawkat I see. In that case, we have to use the localStorage.getItem. We thought of preserving only the data. Thanks

File: script.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  const retrivedData = localStorage.getItem('storedBooks');
+  const recoverdBooks = JSON.parse(retrivedData);
+  collection.innerHTML = '';
+  if (recoverdBooks !== '') {
+    books = recoverdBooks;
+  }
+  for (let j = 0; j < books.length; j += 1) {
+    books[j].bookId = j;
+    const div = document.createElement('div');
+    div.innerHTML = `<p>${books[j].bookTitle}</p>
```
Comment (Wes-Isaac) [inline script.js:25]: - [ ] Great job so far, when I load the page I see an error in the console, which says I can't call a method on a null property, i.e line 22 **books.length**, kindly fix that.

image

File: script.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  title.value = '';
+  author.value = '';
+}
+
+addBtn.addEventListener('click', () => {
+  if (title.value === '' || author.value === '') {
+    alert('Empty Input!');
+  } else {
+    const book = new BookObject(title.value, author.value, i);
+    books.push(book);
```
Comment (Wes-Isaac) [inline script.js:56]: - [ ] Here also I encounter the same problem, can't push to a null property, please fix that.

image

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by keroloussamy — APPROVED** [review]
Hi @mavericks-db, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**